File: TanksGame/Assets/Tanks/Game Scripts/MPGame/ProjectileEvent.cs
[System.Serializable]
public class ProjectileEvent
{
    // Player info
    public long playerId;

    // Starting position and trajectory
    public float originX;
    public float originY;
    public float angle;

    // Projectile properties
    public float speed;
    public float damage;
    public string projectileType;
    
    public long timestamp;
}

File: tanks_backend/src/main/java/hr/antitalent/tanks_backend/websocket/game/ProjectileEvent.java
package hr.antitalent.tanks_backend.websocket.game;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProjectileEvent {
    // Player info
    private String playerId;

    // Starting position and trajectory
    private double originX; 
    private double originY;
    private double angle;

    // Projectile properties
    private float speed;
    private float damage;
    private String projectileType;
    
    private long timestamp;
}
